stable/ambassador/templates/deployment.yaml

```diff
@@ -0,0 +1,147 @@
+{{- if not .Values.daemonSet -}}
```

Comment:
Not sure, maybe @richarddli or @kflynn can give examples (or [docs](https://www.getambassador.io/reference/running/#running-as-daemonset))

I suppose it could be merged to one template since only difference is type and replica count.

Reply:
> In manifest files kind: Deployment needs to be updated to kind: DaemonSet and replicas should be removed in specsection.

Maybe it can be combined into one but with above recommendations from link?

Reply:
I think the Kubernetes docs summarize this best:

```
DaemonSets are similar to Deployments in that they both create Pods, and those Pods have processes which are not expected to terminate (e.g. web servers, storage servers).

Use a Deployment for stateless services, like frontends, where scaling up and down the number of replicas and rolling out updates are more important than controlling exactly which host the Pod runs on. Use a DaemonSet when it is important that a copy of a Pod always run on all or certain hosts, and when it needs to start before other Pods.
```

In practice, we've seen good arguments for both. Our general default is a Deployment, though, and that seems to work well.